spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/config/security/OAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,204 @@
+import org.springframework.security.web.util.matcher.AnyRequestMatcher;
+
+/**
+ * Setup Spring Security OAuth for the Rest Endpoints of Spring Cloud Data Flow.
+ *
+ * @author Gunnar Hillert
+ * @author Ilayaperumal Gopinathan
+ */
+@EnableOAuth2Client
+@EnableWebSecurity
+@Configuration
+@Conditional(OnSecurityEnabledAndOAuth2Enabled.class)
+public class OAuthSecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+	private static final org.slf4j.Logger logger = LoggerFactory.getLogger(OAuthSecurityConfiguration.class);
+
+	@Autowired
+	private SecurityStateBean securityStateBean;
+
+	@Autowired
+	private SecurityProperties securityProperties;
+
+	@Autowired
+	private OAuth2ClientContext oauth2ClientContext;
+
+	@Autowired
+	private AuthorizationCodeResourceDetails authorizationCodeResourceDetails;
+
+	@Autowired
+	private ResourceServerProperties resourceServerProperties;
+
+	@Autowired
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	@Autowired
+	private AuthorizationProperties authorizationProperties;
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+
+		final BasicAuthenticationEntryPoint basicAuthenticationEntryPoint = new BasicAuthenticationEntryPoint();
+		basicAuthenticationEntryPoint.setRealmName(securityProperties.getBasic().getRealm());
+		basicAuthenticationEntryPoint.afterPropertiesSet();
+		final Filter oauthFilter = oauthFilter();
+		BasicAuthenticationFilter basicAuthenticationFilter = new BasicAuthenticationFilter(
+				providerManager(), basicAuthenticationEntryPoint);
+		http.addFilterAfter(oauthFilter, basicAuthenticationFilter.getClass());
+		http.addFilterBefore(basicAuthenticationFilter, oauthFilter.getClass());
+		http.addFilterBefore(oAuth2AuthenticationProcessingFilter(), basicAuthenticationFilter.getClass());
+		this.authorizationProperties.getAuthenticatedPaths().add("/");
+		this.authorizationProperties.getAuthenticatedPaths().add("/api/**");
+		this.authorizationProperties.getAuthenticatedPaths().add("/api");
+		this.authorizationProperties.getAuthenticatedPaths().add(dashboard("/**"));
+		this.authorizationProperties.getAuthenticatedPaths().add(this.authorizationProperties.getDashboardUrl());
+		this.authorizationProperties.getPermitAllPaths().add(this.authorizationProperties.getDashboardUrl());
+		this.authorizationProperties.getPermitAllPaths().add(dashboard("/**"));
+
+		ExpressionUrlAuthorizationConfigurer<HttpSecurity>.ExpressionInterceptUrlRegistry security =
+
+				http.authorizeRequests()
+						.antMatchers(this.authorizationProperties.getPermitAllPaths().toArray(new String[0]))
+						.permitAll()
+						.antMatchers(this.authorizationProperties.getAuthenticatedPaths().toArray(new String[0]))
+						.authenticated();
+
+		security = SecurityConfigUtils.configureSimpleSecurity(security, this.authorizationProperties);
+		security.anyRequest().denyAll();
+		this.securityStateBean.setAuthorizationEnabled(true);
```

Comment:
We can still use `authorizationProperties` and enable/disable based on the value.